### PR TITLE
Update documentation for scale_down and resize_short in python/mxnet/image.py

### PR DESCRIPTION
**Pull request by jhwei**
@mli, @zackchase, @madjam, @nswamy please see if it's OK. There is something wrong with my original branch, so I made some changes.

### CODE REVIEW

File: python/mxnet/image.py [hunk 1]
```diff
@@ -60,7 +60,33 @@ def imdecode(buf, **kwargs):
 
 
 def scale_down(src_size, size):
-    """Scale down crop size if it's bigger than image size."""
+    """Scale down crop size if it's larger than image size.
```
Comment (madjam) [inline python/mxnet/image.py]: Scale -> Scales

File: python/mxnet/image.py [hunk 2]
```diff
@@ -60,7 +60,33 @@ def imdecode(buf, **kwargs):
...
+
+    If width/height of the crop is larger than the width/height of the image,
+    set the width/height to the width/height of the image.
```
Comment (madjam) [inline python/mxnet/image.py]: set -> sets

File: python/mxnet/image.py [hunk 3]
```diff
@@ -60,7 +60,33 @@ def imdecode(buf, **kwargs):
...
+
+    First compare and set the height, then compare and set the width.
```
Comment (madjam) [inline python/mxnet/image.py]: This looks like an implementation detail and should be omitted from API level documentation.

File: python/mxnet/image.py [hunk 4]
```diff
@@ -71,7 +97,45 @@ def scale_down(src_size, size):
 
 
 def resize_short(src, size, interp=2):
-    """Resize shorter edge to size."""
+    """Resize shorter edge to size.
```
Comment (madjam) [inline python/mxnet/image.py]: Resize -> Resizes

**Comment by jhwei** [comment]
@madjam I have deleted the implementation detail and made some word changes. Please see if there is anything else needed to be edited.

File: python/mxnet/image.py [hunk 5]
```diff
@@ -71,7 +95,45 @@ def scale_down(src_size, size):
...
+    Resizes the original image by setting the shorter edge to size
+    and setting the longer edge accordingly.
+    Resizing function is called from OpenCV.
+
+    Parameters
+    ----------
+    src : NDArray
+        The original image.
+    size : int
+        The number to be set for the shorter edge.
```
Comment (zackchase) [inline python/mxnet/image.py]: Can you be more specific about what "number" means here

Comment (jhwei) [inline python/mxnet/image.py]: @zackchase 'size' is the input which is an integer to the function, so do you mean I should specify what 'size' is originally from or I should change 'number' to a more appropriate word like  'size' or 'length'?

Comment (jhwei) [inline python/mxnet/image.py]: @zackchase I changed the word 'number' to 'length' in the doc, please see if it satisfies the requirement.

**Pull request merged**